partial-response/src/main/java/com/iluwatar/partialresponse/VideoClientApp.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2017 Gopinath Langote
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.iluwatar.partialresponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The Partial response pattern is a design pattern in which client specifies fields to fetch to serve.
+ * Here {@link VideoClientApp} is playing as client for {@link VideoResource} server.
+ * Client ask for specific fields information in video to server.
+ * <p>
+ * <p>
+ * {@link VideoResource} act as server to serve video information.
+ */
+
+public class VideoClientApp {
+  private static final Logger LOGGER = LoggerFactory.getLogger(VideoClientApp.class);
+
+  /**
+   * Method as act client and request to server for video details.
+   *
+   * @param args program argument.
+   */
+  public static void main(String[] args) throws Exception {
+    Map<Integer, Video> videos = new HashMap<>();
+    videos.put(1, new Video(1, "Avatar", 178, "epic science fiction film", "James Cameron", "English"));
+    videos.put(2, new Video(2, "Godzilla Resurgence", 120, "Action & drama movie|", "Hideaki Anno", "Japanese"));
+    videos.put(3, new Video(3, "Interstellar", 169, "Adventure & Sci-Fi", "Christopher Nolan", "English"));
+    FieldJsonMapper fieldJsonMapper = new FieldJsonMapper();
+    VideoResource videoResource = new VideoResource(fieldJsonMapper, videos);
```

Comment:
This could be one-liner e.g. `VideoResource videoResource = new VideoResource(new FieldJsonMapper(), videos);`

Reply:
@iluwatar : Done this. Thanks for pointing out this.